toolboxes/mri_core/mri_core_girf_correction.cpp

```diff
@@ -1,160 +1,282 @@
 
-    hoNDArray<std::complex<float>> GIRF::load_girf_kernel(const std::string& girf_string) {
-        std::vector<std::vector<std::complex<float>>> temp_array;
+            bool success = Gadgetron::GIRF::parse_girf_text(girf_string.begin(), girf_string.end(), temp_array);
+            if (!success)
+                throw std::runtime_error("Failed to parse girf string");
 
-        bool success = parse_girf_text(girf_string.begin(),girf_string.end(),temp_array);
-        if (!success) throw std::runtime_error("Failed to parse girf string");
+            hoNDArray<std::complex<float>> result(temp_array[0].size(), temp_array.size());
+            auto data_ptr = result.get_data_ptr();
+            for (auto &v : temp_array)
+                for (auto c : v)
+                    *(data_ptr++) = c;
 
-        hoNDArray<std::complex<float>> result(temp_array[0].size(),temp_array.size());
-        auto data_ptr = result.get_data_ptr();
-        for (auto& v: temp_array)
-            for (auto c: v)
-                *(data_ptr++) = c;
-
-
-        return result;
-
-    }
+            return result;
+        }
 
-    hoNDArray<floatd3>
-    GIRF::girf_correct(const hoNDArray<floatd3> &gradients, const hoNDArray<std::complex<float>> &girf_kernel,
-                       const arma::fmat33 &rotation_matrix, float gradient_sampling_time, float girf_sampling_time,
-                       float TE) {
-        return girf_correct_D(gradients,girf_kernel,rotation_matrix,girf_sampling_time,girf_sampling_time,TE);
-    }
+        hoNDArray<floatd3>
+        girf_correct(const hoNDArray<floatd3> &gradients, const hoNDArray<std::complex<float>> &girf_kernel,
+                                          const arma::fmat33 &rotation_matrix, float gradient_sampling_time, float girf_sampling_time,
+                                          float TE)
+        {
+            return girf_correct_D(gradients, girf_kernel, rotation_matrix, girf_sampling_time, girf_sampling_time, TE);
+        }
 
-    hoNDArray<floatd2>
-    GIRF::girf_correct(const hoNDArray<floatd2> &gradients, const hoNDArray<std::complex<float>> &girf_kernel,
-                       const arma::fmat33 &rotation_matrix, float gradient_sampling_time, float girf_sampling_time,
-                       float TE) {
-        return girf_correct_D(gradients,girf_kernel,rotation_matrix,girf_sampling_time,girf_sampling_time,TE);
-    }
+        hoNDArray<floatd2>
+        girf_correct(const hoNDArray<floatd2> &gradients, const hoNDArray<std::complex<float>> &girf_kernel,
+                                          const arma::fmat33 &rotation_matrix, float gradient_sampling_time, float girf_sampling_time,
+                                          float TE)
+        {
+            return girf_correct_D(gradients, girf_kernel, rotation_matrix, gradient_sampling_time, girf_sampling_time, TE);
+        }
 
+        hoNDArray<std::complex<float>> zeropadding(hoNDArray<std::complex<float>> input, int zpadFactor)
```

Comment:
What's wrong with the padding function we have? If you need a highly specific behaviour, that's fine, but keep it internal to the GIRF module.